### PR TITLE
set up fuzzing for apollo-parser

### PR DESCRIPTION
**Pull request by Geal**
following up on https://github.com/apollographql/apollo-rs/pull/115#discussion_r759136264, I set up a fuzzing test (I see @garypen worked on it too in https://github.com/apollographql/apollo-rs/issues/59#issuecomment-960620361 ) and it is already triggering issues in the lexer, that should be easy to address

### CODE REVIEW

**Comment by Geal** [comment]
the interesting thing is that it can find small issues in the lexer easily if we do not provide any existing query in the corpus. In those cases, a fuzzer just tests things and slowly learns how to make a valid file

**Comment by Geal** [comment]
I added some of those issues and their fixes. The fuzzer will keep running into `unwrap()` calls in the lexer, and infinite loop in the parser. It will generate annoying bugs for a while, but it's an easy way to make the project more robust

File: crates/apollo-parser/src/parser/grammar/extensions.rs [hunk 1]
```diff
@@ -14,6 +14,6 @@ pub(crate) fn extensions(p: &mut Parser) {
         Some("union") => union_::union_type_extension(p),
         Some("enum") => enum_::enum_type_extension(p),
         Some("input") => input::input_object_type_extension(p),
-        _ => p.err("A Type System Extension cannot be applied"),
+        _ => p.err_and_pop("A Type System Extension cannot be applied"),
```
Comment (lrlna) [inline crates/apollo-parser/src/parser/grammar/extensions.rs:17]: ohhh good catch!

**Review by lrlna — APPROVED** [review]
Thank you, this is great!

**Pull request merged**